src/test/java/GameFieldTest.java

```diff
@@ -7,29 +7,23 @@
 
 public class GameFieldTest {
 
-    @Test
-    public void shouldNotAddObjectWhenObjectIsOutOfBoundary(){
+    @Test(expected = OutOfBoundaryException.class)
+    public void shouldThrowOutBoundaryExceptionWhenObjectIsOutOfBoundary() throws OutOfBoundaryException, ObjectIntersectionException {
         GameField gameField = new GameField(new Point(10, 10));
 
         gameField.addObject(new GameObject(new Point(9, 9), new Point(2, 2)));
-
-        Collection<GameObject> gameObjects = gameField.getGameObjects();
-        assertEquals(0, gameObjects.size());
     }
 
-    @Test
-    public void shouldNotAddObjectWhenObjectIntersectsOthers(){
+    @Test(expected = ObjectIntersectionException.class)
+    public void shouldThrowObjectIntersectionExceptionWhenObjectIntersectsOthers() throws OutOfBoundaryException, ObjectIntersectionException {
```

Comment:
just shouldThrow, its clear from ```@Test(expected = ObjectIntersectionException.class)```  that ObjectIntersectionException should be thrown

shouldThrowWhenInsertedObjectIntersectsOthers